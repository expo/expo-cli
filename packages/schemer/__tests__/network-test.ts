import Schemer from '../src/index';

const schema = require('./files/schema.json').schema;
const S = new Schemer(schema, { rootDir: './__tests__' });

describe('Remote', () => {
  it('Icon', async () => {
    expect(
      await S.validateIcon(
        'https://upload.wikimedia.org/wikipedia/commons/0/0f/Icon_Pinguin_2_512x512.png'
      )
    ).toEqual(undefined);
  });

  it('Remote icon dimensions correct', async () => {
    const S = new Schemer({
      properties: {
        icon: {
          meta: { asset: true, dimensions: { width: 100, height: 100 } },
        },
      },
    });
    expect(await S.validateIcon('https://httpbin.org/image/png')).toEqual(undefined);
  });

  it('Remote icon dimensions wrong', async () => {
    let didError = false;
    const S = new Schemer(
      {
        properties: {
          icon: {
            meta: {
              asset: true,
              dimensions: { width: 101, height: 100 },
              contentTypePattern: '^image/png$',
            },
          },
        },
      },
      { rootDir: './__tests__' }
    );
    try {
      await S.validateIcon('https://httpbin.org/image/png');
    } catch (e) {
      didError = true;
      expect(e).toBeTruthy();
      expect(e.errors.length).toBe(1);
      expect(
        e.errors.map(validationError => {
          const { stack, ...rest } = validationError;
          return rest;
        })
      ).toMatchSnapshot();
    }

    expect(didError).toBe(true);
  });
});
