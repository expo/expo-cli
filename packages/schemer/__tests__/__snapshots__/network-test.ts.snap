// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Remote Remote icon dimensions wrong 1`] = `
Array [
  Object {
    "data": "https://httpbin.org/image/png",
    "errorCode": "INVALID_DIMENSIONS",
    "fieldPath": "icon",
    "message": "'icon' should have dimensions 101x100, but the file at 'https://httpbin.org/image/png' has dimensions 100x100",
    "meta": Object {
      "asset": true,
      "contentTypePattern": "^image/png$",
      "dimensions": Object {
        "height": 100,
        "width": 101,
      },
    },
    "name": "ValidationError",
  },
]
`;
